src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxUI.java

```diff
@@ -470,12 +470,12 @@ protected Rectangle rectangleForCurrentValue() {
 
             if (comboBox.getComponentOrientation().isLeftToRight()) {
                 return new Rectangle(insets.left, insets.top + midHeight,
-                        width - (insets.left + insets.right + buttonSize) + 4,
+                        width - (insets.left + insets.right + buttonSize) + 3,
                         height - (insets.top + insets.bottom));
             }
             else {
                 return new Rectangle(insets.left + buttonSize, insets.top + midHeight,
-                        width - (insets.left + insets.right + buttonSize) + 4,
+                        width - (insets.left + insets.right + buttonSize) + 3,
```

Comment:
I guess previous to this fix, we were using BasicComboBoxUI#rectangleForCurrentValue() which returns a width on which we add +4 for editor/textField width.
Since the calculation is basically same for width, why was it giving problem now (where only height is modified in the 8054572 fix) and not before this fix. I dont think this test use to fail before this fix was integrated with promoted CI builds

Reply:
Previously, editable comboboxes on macOS were misaligned altogether. The button and textfield were not at the same height (as seen on the "before" screenshots in [https://bugs.openjdk.org/browse/JDK-8054572](https://bugs.openjdk.org/browse/JDK-8054572)). My assumption is that these pixels that would normally overlap are no longer overlapping because of this.